src/socket-proxy/socket-proxyd.c

```diff
@@ -639,6 +697,11 @@ static int run(int argc, char *argv[]) {
         if (r < 0)
                 return log_error_errno(r, "Failed to attach resolver: %m");
 
+        r = sd_event_add_time(context.event, &context.idle_time, CLOCK_MONOTONIC, UINT64_MAX,
+                              0, idle_time_cb, &context);
```

Comment:
Please allocate the event source when actually needed. Preallocating it here with infinite timeout is wasteful (and not very pretty).